tests/suites/os/tests/safe-reboot/index.js

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const request = require('request-promise');
+const SUPERVISOR_PORT = 48484;
+
+module.exports = {
+	title: 'Safe reboot tests',
+	tests: [
+		{
+			title: 'Respect application locks',
+			run: async function(test) {
+				let fd = '';
+				let result = '';
+				fd = await this.worker.executeCommandInHostOS(
+					`touch /tmp/reboot-check && \
+					mkdir -p /tmp/balena-supervisor/services/test/ && \
+					touch /tmp/balena-supervisor/services/test/updates.lock && \
+					exec {FD}</tmp/balena-supervisor/services/test/updates.lock && \
+					(flock -n $FD && sleep infinity) &
+					echo "$FD"`,
+					this.link,
+				)
+				await this.worker.executeCommandInHostOS(
+					'/usr/libexec/safe_reboot &',
+					this.link,
+				)
+				await this.utils.waitUntil(async () => {
+					result = await this.worker.executeCommandInHostOS(
+						'journalctl -n 20 | grep -q -e "/tmp/balena-supervisor/services/test/updates.lock exists"; echo $?',
+						this.link,
+					);
+					return result === '0';
+				},false, 20, 1000);
+				test.is(
+					result,
+					'0',
+					"Safe reboot waiting on application locks"
+				);
+				await this.worker.executeCommandInHostOS(
+					'[[ -f /tmp/reboot-check ]] && echo "pass"',
+					this.link,
+				).then( async (result) => {
```

Comment:
nitpick here, but the use of `then` here is probably not what I'd go for - `await` already gives you the result of `worker.executeCommandInHostOS()`, so using `then` with a callback isn't really needed and adds some bloat to the code. In general I find mixing `await` and `then` quite confusing.